pkgs/applications/science/robotics/mission-planner/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchurl, makeDesktopItem, makeWrapper, unzip, mono6 }:
+
+let
+  pname = "mission-planner";
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    icon = pname;
+    comment = "MissionPlanner GCS & Ardupilot configuration tool";
+    desktopName = "MissionPlanner";
+    genericName = "Ground Control Station";
+  };
+in stdenv.mkDerivation rec {
+  inherit pname;
+  version = "1.3.74";
+
+  src = fetchurl {
+    url = "https://firmware.ardupilot.org/Tools/MissionPlanner/MissionPlanner-${version}.zip";
+    sha256 = "1cgpmsmmnbzw1lwsdafp8yklk1rwc61yf12vc1ahcc6bl7q2385x";
+  };
+
+  nativeBuildInputs = [ makeWrapper mono6 unzip ];
+  sourceRoot = ".";
+
+  AOT_FILES = [ "MissionPlanner.exe" "MissionPlanner.*.dll" ];
+
+  buildPhase = ''
+    for file in $AOT_FILES
+    do
+      mono --aot $file
+    done
+  '';
+
+  installPhase = ''
+    # create target directoryies
```

Comment:
```suggestion
```

Reply:
Why are comments bad?

Reply:
Comments for complex intuitive parts of code are good. On the other hand comments for easy to overlook and understand code are not that useful and easily get out of date. Creating directories with ``mkdir -p`` is a very common thing in nix and does not require a comment.

Reply:
Ok then